sparse/src/KokkosSparse_StaticCrsGraph.hpp

```diff
@@ -0,0 +1,562 @@
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 4.0
+//       Copyright (2022) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Part of Kokkos, under the Apache License v2.0 with LLVM Exceptions.
+// See https://kokkos.org/LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//@HEADER
+
+#ifndef KOKKOSSPARSE_STATICCRSGRAPH_HPP_
+#define KOKKOSSPARSE_STATICCRSGRAPH_HPP_
+
+#include <Kokkos_Core.hpp>
+
+#ifdef KOKKOS_ENABLE_DEPRECATED_CODE_4
+
+#define KOKKOS_IMPL_DO_NOT_WARN_INCLUDE_STATIC_CRS_GRAPH
+#include <Kokkos_StaticCrsGraph.hpp>
+#undef KOKKOS_IMPL_DO_NOT_WARN_INCLUDE_STATIC_CRS_GRAPH
+
+namespace KokkosSparse {
```

Comment:
I would prefer to take out the
```
#ifdef KOKKOS_ENABLE_DEPRECATED_CODE_4
...
#else
```
block that uses Kokkos_StaticCrsGraph.hpp to provide the declarations. It seems like it will cause deprecated warnings without any benefit. If the user is including this new header, we might as well use the new declarations in it.

Reply:
I like the idea of avoiding the deprecated branch, but could there be a case for example with packages Trilinos (and deprecated code enabled) that pull in `Kokkos_StaticCrsGraph.hpp` indirectly through kokkos-kernels (including KokkosSparse_CrsMatrix.hpp prior to this PR) and use the deprecated functionality? Keeping the guarded deprecated region could prevent breakage in that case, though I'm unsure how likely an issue would be